aesara/link/basic.py

```diff
@@ -323,16 +359,25 @@ def accept(self, fgraph, no_recycling=None, profile=None):
         self.no_recycling = no_recycling
         return self
 
-    def make_all(self, input_storage=None, output_storage=None, storage_map=None):
+    def make_all(
+        self,
+        input_storage: Optional[Any] = None,
+        output_storage: Optional[OutputStorageType] = None,
+        storage_map: Optional[StorageMapType] = None,
+    ) -> Union[
+        Tuple[Callable, List[Container], List[Container], List[Callable], List[Apply]],
+        Tuple[Callable, List[Container], List[Container], List[Any], List[Any]],
+    ]:
```

Comment:
This looks like it could use a name and type of its own.

That change isn't necessary to merge this, but&mdash;given the size of that composite type&mdash;it makes sense.